### PR TITLE
Changed Box to add border='between'

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Changed Box to add border='between'.
This makes it easy to add visual separators between children of Box.

#### Where should the reviewer start?

Box/doc.js

#### What testing has been done on this PR?

Enhanced Box Border story

#### How should this be manually tested?

Storybook

#### Do the grommet docs need to be updated?

yes

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

File: src/js/components/Box/README.md [hunk 1]
```diff
@@ -279,6 +279,7 @@ right
 horizontal
 vertical
 all
+between
 {
   color: 
     string
```
Comment (negreirosleo) [inline src/js/components/Box/README.md]: If it has the possibility of `border.side === between` , I think would be a good Idea to add it here aswell

Comment (negreirosleo) [reply src/js/components/Box/README.md]:
```
....
all
between
{
  color: 
    string
    {
      dark: string,
      light: string
    },
  side: 
    top
    left
    bottom
    right
    horizontal
    vertical
    all
    between
  size: 
...
```

Comment (ShimiSun) [reply src/js/components/Box/README.md]: https://github.com/grommet/grommet/blob/a3c75c2310ac789120b137f6310a432ffb98fb92/src/js/components/Box/doc.js#L40

File: src/js/components/Box/doc.js [hunk 2]
```diff
@@ -168,10 +168,11 @@ export const doc = Box => {
         'horizontal',
         'vertical',
         'all',
+        'between',
       ]),
       BORDER_SHAPE,
       PropTypes.arrayOf(BORDER_SHAPE),
-    ]).description('Include a border.'),
+    ]).description('Include a border. You must have "gap" to use "between"'),
```
Comment (ShimiSun) [inline src/js/components/Box/doc.js]: I wonder if we can add some explanation about `between`, something like 'between will place a border between child elements' or '...where a gap is applied'.

File: src/js/components/Box/README.md (hunk 1 above)
Comment (ericsoderberghp) [reply src/js/components/Box/README.md]: agreed

File: src/js/components/Box/doc.js (hunk 2 above)
Comment (ericsoderberghp) [reply src/js/components/Box/doc.js]: agreed, done

**Pull request merged**